src/otx/algo/callbacks/adaptive_early_stopping.py

```diff
@@ -0,0 +1,85 @@
+    ):
+        """EarlyStoppingWithWarmup callback.
+
+        Args:
+            monitor (str): The metric to monitor.
+            min_delta (float, optional): Minimum change in the monitored quantity
+                to qualify as an improvement. Defaults to 0.0.
+            patience (int, optional): Number of epochs with no improvement
+                after which training will be stopped. Defaults to 3.
+            verbose (bool, optional): If True, prints messages to stdout. Defaults to False.
+            mode (str, optional): One of {"min", "max"}. In "min" mode, training will stop when
+                the quantity monitored has stopped decreasing. In "max" mode,
+                it will stop when the quantity monitored has stopped increasing. Defaults to "min".
+            strict (bool, optional): If True, the monitored quantity must improve
+                according to the mode for it to be considered an improvement.
+                Defaults to True.
+            check_finite (bool, optional): If True, check that the monitored quantity is
+                finite before considering an improvement. Defaults to True.
+            stopping_threshold (float | None, optional): The threshold to stop training.
+                Defaults to None.
+            divergence_threshold (float | None, optional): The threshold for divergence detection.
+                Defaults to None.
+            check_on_train_epoch_end (bool | None, optional): If True,
+                checks the stopping criterion on train_epoch_end. Defaults to None.
+            log_rank_zero_only (bool, optional): If True, logs should only be printed from rank 0.
+                Defaults to False.
+            warmup_iters (int, optional): Number of warmup iterations. Defaults to 100.
+            warmup_epochs (int, optional): Number of warmup epochs. Defaults to 3.
+        """
+        super().__init__(
+            monitor=monitor,
+            min_delta=min_delta,
+            patience=patience,
+            verbose=verbose,
+            mode=mode,
+            strict=strict,
+            check_finite=check_finite,
+            stopping_threshold=stopping_threshold,
+            divergence_threshold=divergence_threshold,
+            check_on_train_epoch_end=check_on_train_epoch_end,
+            log_rank_zero_only=log_rank_zero_only,
+        )
+        # two thresholds to have invariant to extra small datasets and larger datasets
+        self.warmup_iters = warmup_iters
+        self.warmup_epochs = warmup_epochs
+
+    def _should_skip_check(self, trainer: pl.Trainer) -> bool:
+        from lightning.pytorch.trainer.states import TrainerFn
+
+        current_iter = trainer.current_epoch * trainer.num_training_batches
+        warmup_threshold = max(self.warmup_epochs * trainer.num_training_batches, self.warmup_iters)
+        return trainer.state.fn != TrainerFn.FITTING or trainer.sanity_checking or current_iter < warmup_threshold
```

Comment:
```suggestion
        return super()._should_skip_check(trainer) or current_iter < warmup_threshold
```
How about using parent class's function? It can reflect changes if code in lightning side is updated later.